Clapet/Views/Settings/SettingsView.swift

```diff
@@ -23,9 +23,10 @@ struct SettingsView: View {
             AdvancedSettings().tabItem {
                 Label("advanced", systemImage: "ellipsis.curlybraces")
             }
-        }.onReceive(NotificationCenter.default.publisher(for: NSWindow.willCloseNotification)) { _ in
-            AppDelegate.hideApplication()
         }
+//        .onReceive(NotificationCenter.default.publisher(for: NSWindow.willCloseNotification)) { _ in
+//            AppDelegate.hideApplication()
+//        }
```

Comment:
This should be behind a condition on `showDockIcon` (if it's true we do nothing, else we hide), because if the option is not enabled, the dock icon would still stay after opening and closing settings

Reply:
I see why you did that, for some reason a willCloseNotification event is fired on a TUINSWindow when the settings are opened, I don't know what this window is, and this is beyond my (low) swiftui skills, but since the TUINSWindow is not visible, we can fix the settings not opening by filtering the events on isVisible :

```
.onReceive(NotificationCenter.default.publisher(for: NSWindow.willCloseNotification)) { a in
    if let window = a.object {
        if (window as! NSWindow).isVisible {
            AppDelegate.hideApplication()
        }
    }
}
```

You can then also add your condition on `showDockIcon`